homeassistant/components/tuya/climate.py

```diff
@@ -0,0 +1,463 @@
+        ):
+            return True
+        return False
+
+    @property
+    def temperature_unit(self) -> str:
+        """Return true if fan is on."""
+        if self.is_celsius():
+            return TEMP_CELSIUS
+        return TEMP_FAHRENHEIT
+
+    @property
+    def current_temperature(self) -> float | None:
+        """Return the current temperature."""
+        if (
+            DPCODE_TEMP_CURRENT not in self.tuya_device.status
+            and DPCODE_TEMP_CURRENT_F not in self.tuya_device.status
+        ):
+            return None
+
+        if self.is_celsius():
+            return (
+                self.tuya_device.status.get(DPCODE_TEMP_CURRENT, 0)
+                * 1.0
+                / (10 ** self.get_temp_current_scale())
+            )
+        return (
+            self.tuya_device.status.get(DPCODE_TEMP_CURRENT_F, 0)
+            * 1.0
+            / (10 ** self.get_temp_current_scale())
+        )
+
+    @property
+    def current_humidity(self) -> int:
+        """Return the current humidity."""
+        return int(self.tuya_device.status.get(DPCODE_HUMIDITY_CURRENT, 0))
+
+    @property
+    def target_temperature(self) -> float:
+        """Return the temperature currently set to be reached."""
+        return (
+            self.tuya_device.status.get(DPCODE_TEMP_SET, 0)
+            * 1.0
+            / (10 ** self.get_temp_set_scale())
+        )
+
+    @property
+    def max_temp(self) -> float:
+        """Return the maximum temperature."""
+        return self.target_temperature_high
+
+    @property
+    def min_temp(self) -> float:
+        """Return the minimum temperature."""
+        return self.target_temperature_low
+
+    @property
+    def target_temperature_high(self) -> float:
+        """Return the upper bound target temperature."""
+        if self.is_celsius():
+            if DPCODE_TEMP_SET not in self.tuya_device.function:
+                return 0
+            temp_value = json.loads(
+                self.tuya_device.function.get(DPCODE_TEMP_SET, {}).values
+            )
+            return temp_value.get("max", 0) * 1.0 / (10 ** self.get_temp_set_scale())
+        if DPCODE_TEMP_SET_F not in self.tuya_device.function:
+            return 0
+        temp_value = json.loads(
+            self.tuya_device.function.get(DPCODE_TEMP_SET_F, {}).values
+        )
+        return temp_value.get("max", 0) * 1.0 / (10 ** self.get_temp_set_scale())
+
+    @property
+    def target_temperature_low(self) -> float:
+        """Return the lower bound target temperature."""
+        if self.is_celsius():
+            if DPCODE_TEMP_SET not in self.tuya_device.function:
+                return 0
+            temp_value = json.loads(
+                self.tuya_device.function.get(DPCODE_TEMP_SET, {}).values
+            )
+            low_value = (
+                temp_value.get("min", 0) * 1.0 / (10 ** self.get_temp_set_scale())
+            )
+            return low_value
```

Comment:
It looks like we'll return 0 here if the `"min"` item is missing in the temp value dict. That doesn't seem right. If we don't know the state we should return `None`.

Reply:
fixed